packages/storybook/src/generators/configuration/lib/util-functions.ts

```diff
@@ -488,11 +488,14 @@ export function addStorybookToNamedInputs(tree: Tree) {
   }
 }
 
-export function addStorybookToTargetDefaults(tree: Tree) {
+export function addStorybookToTargetDefaults(tree: Tree, setCache = true) {
   const nxJson = readNxJson(tree);
 
   nxJson.targetDefaults ??= {};
   nxJson.targetDefaults['build-storybook'] ??= {};
+  if (setCache) {
```

Comment:
The `cache` option was introduced in Nx 17, but there's a migration for 16.5 using this utility. We don't set it in that case.